giftcompare/users/views.py

```diff
@@ -7,13 +7,17 @@
 from .models import CustomUser
 from .serializers import CustomUserSerializer
 
+from giftcompare.permissions import IsAdminOrSelf, IsAdminOrNothing
 
 class CustomUserList(APIView):
+    permission_classes = [IsAdminOrNothing]
+
     def get(self, request):
         users = CustomUser.objects.all()
         serializer = CustomUserSerializer(users, many=True)
         return Response(serializer.data)
-    
+
+class UserRegistrationView(APIView):
     def post(self, request):
```

Comment:
In this way everyone, including existing users already authenticated, can create a new user.
We need to add a permission that denies to use this endpoint to authenticated users.

Reply:
Added permission to the view

Reply:
Thanks, this is working fine now.